app/controllers/api/v2/qc_files_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    # Provides a JSON API controller for QcFile
+    # See: http://jsonapi-resources.com/ for JSONAPI::Resource documentation
+    class QcFilesController < JSONAPI::ResourceController
+      # By default JSONAPI::ResourceController provides most the standard
+      # behaviour, and in many cases this file may be left empty.
+    end
+
+    class QcFileProcessor < JSONAPI::Processor
```

Comment:
In case you're wondering why I overrode this, it's so that we can get the incoming contents persisted to disk before a new QcFile gets created and we can reference the file in the new QcFile record.  This has to be ready before the new instance is created and the Processor is responsible for asking the `Resource` class to generate the new record.

This varies from the way I did it in `Transfer`s where I did some modifications of the operation parameters and moving data into the context.  I think the approach here is cleaner as it doesn't pollute the context and is makes sure we only do any parsing of inputs for a `create_resource` operation while leaving all the other operations functioning as normal.